### PR TITLE
jaegertracing build/ops integration

### PR DESCRIPTION
**Pull request by ideepika**
## Milestones
- [x] Build Jaeger as External Project ( all cmake at scalable level)
- [x] OSD trace using marker spans
- [ ] Off the wire context transfer configuration
- [ ] Documentation
- [ ]  RGW path trace
- [ ] Use Case Documentation 
- [ ] Reviews, Cleanups and Documentation update
- [ ] Code Coverage and Tests 

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -335,6 +335,19 @@ if(WITH_BLKIN)
   include_directories(SYSTEM src/blkin/blkin-lib)
 endif(WITH_BLKIN)
 
+option(WITH_JAEGER "Use jaeger to create tracepoints and display it on Jaeger UI" ON)
+
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply CMakeLists.txt]: @wjwithagen i think the reason why we want to enable this by default is to help our end user to track down performance issues without rebuilding/resinstalling ceph. 

> again another tracing tool doing its thing.

i am not sure what do you intended to put here. if you are interested in how these tracing tools working together please watch https://www.youtube.com/watch?v=NHoI9F9bZeM .

> So the developer should make his/her own choice for what he wants to ON

that's why an option is added for it.

File: cmake/modules/FindYAMLCPP.cmake [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#.rst:
+# FindYAMLCPP
```
Comment (tchaikov) [inline cmake/modules/FindYAMLCPP.cmake]: please note, we already include seastar as a submodule. which check for yaml-cpp as well. but i understand we should not rely on it as Jaeger and crimson are not hard-wired, and they should not. what i suggest is to use the same name for the package to be found -- "yaml-cpp". and expose the same set of variables / target as https://github.com/scylladb/seastar/blob/master/cmake/Findyaml-cpp.cmake does.

to name some ratinales:

1. no need to check yaml-cpp / YAMLCPP twice. as the variables are cached
2. no divergence when searching for yaml-cpp / YAMLCPP in the same project. we **include** seastar. so it would be confusing when developer finds that we are detecting both yaml-cpp and YAMLCPP.
3. avoid the potential risk of compiling and linking against two different yaml-cpp libraries found by these two scripts.

File: cmake/modules/FindJAEGER.cmake [hunk 3]
```diff
@@ -0,0 +1,112 @@
+#.rst:
```
Comment (tchaikov) [inline cmake/modules/FindJAEGER.cmake]: could you rename `FindJAGER.cmake` to `FindJaeger.cmake`. i think it'd be better to use the exact name used by upstream. and rename the variables and target accordingly?

File: cmake/modules/FindOPENTRACE.cmake [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+#
+# This module finds the `Jaeger`__ library.
+#
+# Imported target
+# ^^^^^^^^^^^^^^^
+#
+# This module defines the following :prop_tgt:`IMPORTED` target:
+#
+# ``Jaeger``
+#   The jaeger library, if found
```
Comment (tchaikov) [inline cmake/modules/FindOPENTRACE.cmake]: the document is not updated.

File: cmake/modules/FindOPENTRACE.cmake [hunk 5]
```diff
@@ -0,0 +1,112 @@
+#.rst:
```
Comment (tchaikov) [inline cmake/modules/FindOPENTRACE.cmake]: could you rename the file to `FindOpenTracing.cmake` and rename the variables and target accordingly?

**Comment by ideepika** [comment]
@tchaikov @wjwithagen can you help to review this pr?

**Comment by ideepika** [comment]
jenkins test make check

**Comment by ideepika** [comment]
jenkins test make check

**Comment by ideepika** [comment]
jenkins test make check

**Comment by ideepika** [comment]
@tchaikov builds passing: https://shaman.ceph.com/builds/ceph/wip-jaegertracing-in-ceph/0af1bc815804b8d63488e82884dc913bd8509fd1/
please take a look when you find time  :)
thanks!

File: CMakeLists.txt [hunk 6]
```diff
@@ -354,6 +354,12 @@ if(WITH_BLKIN)
   include_directories(SYSTEM src/blkin/blkin-lib)
 endif(WITH_BLKIN)
 
+option(WITH_JAEGER "Use jaeger to create tracepoints and display it on Jaeger
+UI" ON)
```
Comment (tchaikov) [inline CMakeLists.txt]: how is this option connected with "display it on Jaeger UI"? do we have a builtin Jaeger UI support?

File: ceph.spec.in [hunk 7]
```diff
@@ -90,6 +91,8 @@
 # disable dwz which compresses the debuginfo
 %global _find_debuginfo_dwz_opts %{nil}
 
+# for jaeger testing
+%define _unpackaged_files_terminate_build 0
```
Comment (tchaikov) [inline ceph.spec.in:95]: shall we drop this setting?

File: ceph.spec.in [hunk 8]
```diff
@@ -200,6 +203,18 @@ BuildRequires:	socat
 %if 0%{with zbd}
 BuildRequires:  libzbd-devel
 %endif
+%if 0%{with jaeger}
+BuildRequires:	bison
```
Comment (tchaikov) [inline ceph.spec.in:207]: are you trying to align with some `BuildRequires`?

File: ceph.spec.in [hunk 9]
```diff
@@ -200,6 +203,18 @@ BuildRequires:	socat
...
+BuildRequires:	flex
+%if 0%{?fedora} || 0%{?rhel}
+BuildRequires: json-devel
```
Comment (tchaikov) [inline ceph.spec.in:210]: ditto.

File: ceph.spec.in [hunk 10]
```diff
@@ -921,7 +942,35 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
 Obsoletes:	libcephfs2-devel < %{_epoch_prefix}%{version}-%{release}
 %description -n libcephfs-devel
 This package contains libraries and headers needed to develop programs
-that use Cephs distributed file system.
+that use Ceph distributed file system.
+
+%if 0%{with jaeger}
+%package -n libjaeger
+Summary:	Ceph distributed file system client library
```
Comment (tchaikov) [inline ceph.spec.in:949]: this does not sound right.

File: ceph.spec.in [hunk 11]
```diff
@@ -921,7 +942,35 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
 Obsoletes:	libcephfs2-devel < %{_epoch_prefix}%{version}-%{release}
 %description -n libcephfs-devel
 This package contains libraries and headers needed to develop programs
-that use Cephs distributed file system.
+that use Ceph distributed file system.
+
+%if 0%{with jaeger}
```
Comment (tchaikov) [inline ceph.spec.in:947]: could you expand a little bit in the commit message?

File: ceph.spec.in [hunk 12]
```diff
@@ -921,7 +942,35 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
...
+%package -n libjaeger
+Summary:	Ceph distributed file system client library
+%if 0%{?suse_version}
+Group:		System/Libraries
+%endif
+Obsoletes:	libjaeger < %{_epoch_prefix}%{version}-%{release}
```
Comment (tchaikov) [inline ceph.spec.in:953]: why?

File: ceph.spec.in [hunk 13]
```diff
@@ -921,7 +942,35 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
...
+Obsoletes:	libthrift < 0.13.0
```
Comment (tchaikov) [inline ceph.spec.in:954]: why?

File: ceph.spec.in [hunk 14]
```diff
@@ -921,7 +942,35 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
...
+Provides: 	libjaegertracing.so.0()(64bit)
+Provides:	libopentracing.so.1()(64bit)
+Provides:	libthrift.so.0.13.0()(64bit)
+%description -n libjaeger
+This package contains libraries and headers needed to provide distributed
+tracing for Ceph.
+
+%package -n libjaeger-devel
```
Comment (tchaikov) [inline ceph.spec.in:962]: who will be consuming this package?

File: ceph.spec.in [hunk 15]
```diff
@@ -921,7 +942,35 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
...
+Summary:	Ceph distributed file system headers
+%if 0%{?suse_version}
+Group:		Development/Libraries/C and C++
+%endif
+Requires:	libjaeger = %{_epoch_prefix}%{version}-%{release}
+Obsoletes:	libjaeger-devel < %{_epoch_prefix}%{version}-%{release}
+Provides:	libjaeger-devel = %{_epoch_prefix}%{version}-%{release}
```
Comment (tchaikov) [inline ceph.spec.in:969]: why?

File: ceph.spec.in [hunk 16]
```diff
@@ -935,7 +984,7 @@ Requires:	python%{python3_pkgversion}-ceph-argparse = %{_epoch_prefix}%{version}
 Provides:	python-cephfs = %{_epoch_prefix}%{version}-%{release}
 Obsoletes:	python-cephfs < %{_epoch_prefix}%{version}-%{release}
 %description -n python%{python3_pkgversion}-cephfs
-This package contains Python 3 libraries for interacting with Cephs distributed
+This package contains Python 3 libraries for interacting with Ceph's distributed
```
Comment (tchaikov) [inline ceph.spec.in:987]: nit, i'd suggest move this change into a separate commit.

File: ceph.spec.in [hunk 17]
```diff
@@ -921,7 +942,35 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
...
+
+%if 0%{with jaeger}
+%package -n libjaeger
+Summary:	Ceph distributed file system client library
+%if 0%{?suse_version}
+Group:		System/Libraries
+%endif
+Obsoletes:	libjaeger < %{_epoch_prefix}%{version}-%{release}
+Obsoletes:	libthrift < 0.13.0
+Provides: 	libjaegertracing.so.0()(64bit)
```
Comment (tchaikov) [inline ceph.spec.in:955]: who will be consuming for instance `libjaegertracing.so.0`?

File: cmake/modules/BuildBoost.cmake [hunk 18]
```diff
@@ -171,6 +171,7 @@ function(do_build_boost version)
   include(ExternalProject)
   ExternalProject_Add(Boost
     ${source_dir}
+    INSTALL_DIR "external"
```
Comment (tchaikov) [inline cmake/modules/BuildBoost.cmake:174]: why?

File: cmake/modules/BuildJaeger.cmake [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+#License version 2.1, as published by the Free Software
+#Foundation.  See file COPYING.
+################################################################################
+#
+# This module builds Jaeger after it's dependencies are installed and discovered
+# opentracing: is built using cmake/modules/Buildopentracing.cmake
+# Thrift: build using cmake/modules/Buildthrift.cmake
+# yaml-cpp, nlhomann-json: are installed locally and then discovered using
+# Find<package>.cmake
+# Boost Libraries used for building thrift are build and provided by
```
Comment (tchaikov) [inline cmake/modules/BuildJaeger.cmake:17]: do you mean the thrift does not compile with the boost libraries shipped by distro?

File: cmake/modules/BuildJaeger.cmake [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+                        -DHUNTER_ENABLED=OFF
+                        -DBUILD_TESTING=OFF
+                        -DJAEGERTRACING_BUILD_EXAMPLES=OFF
+                        -DCMAKE_PREFIX_PATH=${CMAKE_BINARY_DIR}/external
+                        -DCMAKE_INSTALL_RPATH=${CMAKE_BINARY_DIR}/external
+			-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
+                        -Dopentracing_DIR=${CMAKE_SOURCE_DIR}/src/jaegertracing/opentracing-cpp
+			-Dnlohmann_json_DIR=/usr/lib
+			-DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/external
+			-DCMAKE_FIND_ROOT_PATH="${CMAKE_SOURCE_DIR}/debian/tmp${CMAKE_BINARY_DIR}/external;${CMAKE_BINARY_DIR}/external"
```
Comment (tchaikov) [inline cmake/modules/BuildJaeger.cmake:37]: how does "debian" kick in?

File: CMakeLists.txt (hunk 6 above)
Comment (ideepika) [reply CMakeLists.txt]: jaeger provides a docker image with agent, collector and frontend. So, in this case Ceph will generate the spans, while an external agent will work on sending them to a collector. which will then render the span to Jaeger UI.

File: ceph.spec.in (hunk 8 above)
Comment (tchaikov) [reply ceph.spec.in:207]: by "align" i mean the spacing here looks weird.
```
BuildRequires:  libzbd-devel
BuildRequires:	bison
```

File: CMakeLists.txt (hunk 6 above)
Comment (tchaikov) [reply CMakeLists.txt]: Jaeger UI is not part of what is implemented using WITH_JAEGER. it can be a part of the description of the docker image you are referencing though.

File: ceph.spec.in (hunk 7 above)
Comment (ideepika) [reply ceph.spec.in]: I am afraid we might not have any other way around, since we are building jaeger dependencies with their header files as a dependency too( only needed during build time ) we are not packaging them, which without this option leads to build failure.

File: ceph.spec.in (hunk 17 above)
Comment (ideepika) [reply ceph.spec.in:952]: the internals of Provide script was not able to discover `libjaegertracing.so.0` explicitly, and hence all the packages that had libjaeger as a dependency were failing with missing dependencies, explicit indication resolved that issue

File: cmake/modules/BuildBoost.cmake (hunk 18 above)
Comment (ideepika) [reply cmake/modules/BuildBoost.cmake:174]: I tried keeping it build only, but jaeger dependencies needed Boost installed files during build time, distro package boost lead to interfare with Ceph build boost, so I instead went with installing Ceph Boost altogether.

File: cmake/modules/BuildJaeger.cmake (hunk 19 above)
Comment (ideepika) [reply cmake/modules/BuildJaeger.cmake:15]: yes, for some distro like ubuntu the older version of Boost does not support the latest thrift. ( we can update this once we drop support for bionic)

File: cmake/modules/BuildJaeger.cmake (hunk 20 above)
Comment (ideepika) [reply cmake/modules/BuildJaeger.cmake]: updated, should work without it, testing to verify

File: ceph.spec.in (hunk 10 above)
Comment (ideepika) [reply ceph.spec.in:947]: updated

File: ceph.spec.in (hunk 16 above)
Comment (ideepika) [reply ceph.spec.in:971]: done.

File: ceph.spec.in (hunk 12 above)
Comment (ideepika) [reply ceph.spec.in:951]: i agree, since it's not a active devel package, removed

File: ceph.spec.in (hunk 11 above)
Comment (ideepika) [reply ceph.spec.in:945]: I added comment on --without-jaeger build option

**Comment by ideepika** [comment]
@tchaikov I removed the `do not terminate centos build on failure` option from spec file. Here's successful build and pulpito run with jaeger enabled:
https://shaman.ceph.com/repos/ceph/wip-jaegertracing-in-ceph/84ab4af6ede2e93afa50b24b7bd679f27c179a58/
https://pulpito.ceph.com/ideepika-2020-10-23_07:24:38-rados:cephadm-wip-jaegertracing-in-ceph-distro-basic-smithi/

I'll be happy to have your feedback, thanks!

File: ceph.spec.in (hunk 7 above)
Comment (tchaikov) [reply ceph.spec.in]: please remove those header files in `%install` then.

File: ceph.spec.in (hunk 8 above)
Comment (tchaikov) [reply ceph.spec.in]: this comment is not addressed.

File: ceph.spec.in (hunk 17 above)
Comment (tchaikov) [reply ceph.spec.in:951]: could you be more specific which packages have libjaeger as a dependency?

File: cmake/modules/BuildBoost.cmake (hunk 18 above)
Comment (tchaikov) [reply cmake/modules/BuildBoost.cmake:174]: which jaeger dependencies need Boost installed files during build time in what way?

File: cmake/modules/BuildJaeger.cmake (hunk 19 above)
Comment (tchaikov) [reply cmake/modules/BuildJaeger.cmake:15]: boost 1.72 is required since 713efb66e8d7b2585dfd6783f8f23ce409b50871. so i guess it's new enough?

Comment (ideepika) [reply cmake/modules/BuildJaeger.cmake:15]: @tchaikov I rebased and am trying to eliminate installing boost altogether as well. hitting this issue on jenkins for opensuse and focal:
```
build/ceph-16.0.0-6677-gea00988b/obj-x86_64-linux-gnu/external/include/thrift/transport/TTransportException.h:23:10: fatal error: boost/numeric/conversion/cast.hpp: No such file or directory
   23 | #include <boost/numeric/conversion/cast.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I will update once i can reproduce this and fix it.

File: cmake/modules/BuildBoost.cmake (hunk 18 above)
Comment (ideepika) [reply cmake/modules/BuildBoost.cmake:174]: thrift does and when I did build alone, i still can see include files generated, but when I tried compiling without install I am getting: https://github.com/ceph/ceph/pull/31358/files#r513359062 locally no issues seen in build so I suppose it's with the package time path configuration, I will test it today and check what could be done

File: cmake/modules/BuildJaeger.cmake (hunk 19 above)
Comment (ideepika) [reply cmake/modules/BuildJaeger.cmake:15]: @tchaikov resolved this issue updated code to remove installing boost, it was issue with Jaeger not including Boost headers.
https://shaman.ceph.com/builds/ceph/wip-jaegertracing-in-ceph/

File: src/osd/PrimaryLogPG.cc [hunk 21]
```diff
@@ -2244,6 +2256,7 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
 	get_osdmap()->require_osd_release >= ceph_release_t::kraken) {
       record_write_error(op, oid, nullptr, r);
     } else {
+      //TODO: jaeger: add span
```
Comment (jdurgin) [inline src/osd/PrimaryLogPG.cc:2259]: don't think we need a span for returning an error to the client

File: src/osd/OSD.cc [hunk 22]
```diff
@@ -7017,6 +7020,11 @@ void OSD::dispatch_session_waiting(const ceph::ref_t<Session>& session, OSDMapRe
 
 void OSD::ms_fast_dispatch(Message *m)
 {
+
+#ifdef HAVE_JAEGER jaeger_tracing::init_tracer("osd-services-reinit");
```
Comment (jdurgin) [inline src/osd/OSD.cc:7024]: nit: HAVE_JAEGER check should be on a line by itself

File: src/osd/OSD.cc [hunk 23]
```diff
@@ -7077,7 +7085,13 @@ void OSD::ms_fast_dispatch(Message *m)
     tracepoint(osd, ms_fast_dispatch, reqid.name._type,
         reqid.name._num, reqid.tid, reqid.inc);
   }
-
+#ifdef HAVE_JAEGER
+  op->set_osd_parent_span(dispatch_span);
+  if(op->osd_parent_span){
```
Comment (jdurgin) [inline src/osd/OSD.cc:7090]: nit: all the ifs should be followed by a space, i.e. 
```suggestion
  if (op->osd_parent_span){
```

File: src/osd/OpRequest.h [hunk 24]
```diff
@@ -88,7 +91,16 @@ struct OpRequest : public TrackedOp {
   epoch_t min_epoch = 0;      ///< min epoch needed to handle this msg
 
   bool hitset_inserted;
-
+#ifdef HAVE_JAEGER
+  jspan osd_parent_span = nullptr;
+  void set_osd_parent_span(jspan& span) {
+    if(osd_parent_span){
```
Comment (jdurgin) [inline src/osd/OpRequest.h:97]: spacing - should have a space before {

File: src/osd/OpRequest.h [hunk 25]
```diff
@@ -88,7 +91,16 @@ struct OpRequest : public TrackedOp {
...
+      jaeger_tracing::finish_span(osd_parent_span);
+    }
+    osd_parent_span = move(span); }
```
Comment (jdurgin) [inline src/osd/OpRequest.h:100]: } should be on the next line

File: src/osd/OSD.cc [hunk 26]
```diff
@@ -9627,6 +9641,17 @@ void OSD::enqueue_op(spg_t pg, OpRequestRef&& op, epoch_t epoch)
...
   op->osd_trace.keyval("priority", priority);
   op->osd_trace.keyval("cost", cost);
+#ifdef HAVE_JAEGER
+  if(op->osd_parent_span){
+    auto enqueue_span = jaeger_tracing::child_span(__func__, op->osd_parent_span);
+    enqueue_span->Log({
+	{"priority", priority},
+	{"cost", cost},
+	{"epoch", epoch},
+	{"owner", owner} //Not got owner in UI
```
Comment (jdurgin) [inline src/osd/OSD.cc:9651]: type would be helpful as well

File: src/vstart.sh [hunk 27]
```diff
@@ -1550,6 +1552,44 @@ if [ "$CEPH_NUM_RGW" -gt 0 ]; then
...
+       fi
+       if [ ! "$(podman ps -aq -f name=jaeger)" ]; then
+         $service "$@"
+       fi
+     else
+         echo "cannot find docker or podman, please restart service and rerun."
+     fi
+ }
+
+if $with_jaeger; then
```
Comment (jdurgin) [inline src/vstart.sh:1578]: looks like this is missing an option defining it above

**Review by jdurgin — CHANGES_REQUESTED** [review]
looks good in general, a few minor things noted

File: src/osd/OpRequest.h (hunk 25 above)
Comment (ideepika) [reply src/osd/OpRequest.h]: updated.

File: src/osd/OpRequest.h (hunk 24 above)
Comment (ideepika) [reply src/osd/OpRequest.h:97]: updated.

File: src/osd/PrimaryLogPG.cc (hunk 21 above)
Comment (ideepika) [reply src/osd/PrimaryLogPG.cc]: was not sure whether it would be useful, ack :+1:

File: ceph.spec.in [hunk 28]
```diff
@@ -916,7 +935,21 @@ Provides:	libcephfs2-devel = %{_epoch_prefix}%{version}-%{release}
 Obsoletes:	libcephfs2-devel < %{_epoch_prefix}%{version}-%{release}
 %description -n libcephfs-devel
 This package contains libraries and headers needed to develop programs
-that use Cephs distributed file system.
+that use Ceph distributed file system.
```
[the remark this answers is not in the document]
Comment (ideepika) [reply ceph.spec.in:938]: updated to match all instances to `Ceph` using Ceph as a noun since we have been using it in most other descriptions.

File: src/vstart.sh (hunk 27 above)
Comment (ideepika) [reply src/vstart.sh]: yep, missed to one cherry pick, thanks for catching that!

**Comment by ideepika** [comment]
looks green mostly:
http://pulpito.front.sepia.ceph.com/ideepika-2020-12-11_19:11:18-rados-wip-jaegertracing-in-ceph-distro-basic-smithi/

File: CMakeLists.txt [hunk 29]
```diff
@@ -364,6 +364,11 @@ if(WITH_BLKIN)
   include_directories(SYSTEM src/blkin/blkin-lib)
 endif(WITH_BLKIN)
 
+option(WITH_JAEGER "Enable jaegertracing and it's dependent libraries" OFF)
```
Comment (ronen-fr) [inline CMakeLists.txt:367]: its

File: cmake/modules/BuildJaeger.cmake [hunk 30]
```diff
@@ -0,0 +1,65 @@
+# This module builds Jaeger after it's dependencies are installed and discovered
```
Comment (ronen-fr) [inline cmake/modules/BuildJaeger.cmake:1]: its

File: src/vstart.sh [hunk 31]
```diff
@@ -1550,6 +1558,46 @@ if [ "$CEPH_NUM_RGW" -gt 0 ]; then
...
+     else
+         echo "cannot find docker or podman, please restart service and rerun."
+     fi
+ }
+
+echo ""
+if [ $with_jaeger -eq 1 ]; then
+    debug echo "Enabling jaegertracing..."
+    docker_service run -d --name jaeger \
+  -e COLLECTOR_ZIPKIN_HTTP_PORT=9411 \
```
Comment (ronen-fr) [inline src/vstart.sh:1588]: DO we really need all these interfaces from the example configuration?

Comment (ideepika) [reply src/vstart.sh:1588]: @ronen-fr you are right, this we can avoid, and even we are not using zipkin traces, i will update it, thanks for the feedback!

**Review by jdurgin — APPROVED** [review]
Looks good! let's merge this and handle follow-on work in further PRs.

**Comment by tchaikov** [comment]
@ideepika @jdurgin this change breaks the build
```
-- Checking for one of the modules 'yaml-cpp'
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find yaml-cpp: Found unsuitable version "", but required is at
  least "0.5.1" (found yaml-cpp_LIBRARY-NOTFOUND)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:376 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/Findyaml-cpp.cmake:46 (find_package_handle_standard_args)
  src/CMakeLists.txt:331 (_find_package)
  src/seastar/cmake/SeastarDependencies.cmake:98 (find_package)
  src/seastar/CMakeLists.txt:330 (seastar_find_dependencies)


-- Configuring incomplete, errors occurred!
See also "/home/jenkins-build/build/workspace/ceph-pull-requests/build/CMakeFiles/CMakeOutput.log".
See also "/home/jenkins-build/build/workspace/ceph-pull-requests/build/CMakeFiles/CMakeError.log".
+ exit 1
```

**Comment by tchaikov** [comment]
@ideepika 
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes

we don't have a component named "core" or "jaeger". would you please be more specific in future?

**Comment by ideepika** [comment]
> @ideepika @jdurgin this change breaks the build
> 
> ```
> -- Checking for one of the modules 'yaml-cpp'
> CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
>   Could NOT find yaml-cpp: Found unsuitable version "", but required is at
>   least "0.5.1" (found yaml-cpp_LIBRARY-NOTFOUND)
> Call Stack (most recent call first):
>   /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:376 (_FPHSA_FAILURE_MESSAGE)
>   cmake/modules/Findyaml-cpp.cmake:46 (find_package_handle_standard_args)
>   src/CMakeLists.txt:331 (_find_package)
>   src/seastar/cmake/SeastarDependencies.cmake:98 (find_package)
>   src/seastar/CMakeLists.txt:330 (seastar_find_dependencies)
> 
> 
> -- Configuring incomplete, errors occurred!
> See also "/home/jenkins-build/build/workspace/ceph-pull-requests/build/CMakeFiles/CMakeOutput.log".
> See also "/home/jenkins-build/build/workspace/ceph-pull-requests/build/CMakeFiles/CMakeError.log".
> + exit 1
> ```

@tchaikov seems like missed to catch it, i will fix it asap, thanks!

**Comment by ideepika** [comment]
> @ideepika
> could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes
> 
> we don't have a component named "core" or "jaeger". would you please be more specific in future?

thanks, will keep in mind!

**Comment by tchaikov** [comment]
FTBFS addressed by #38609
